addons/mass_mailing/models/mail_notification.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class Notification(models.Model):
+    # Extend the mail.notification model for tracking (marketing purpose)
+    _inherit = 'mail.notification'
+
+    display_name = fields.Char(compute='_compute_display_name')
+    # mail data
+    mail_id_int = fields.Integer(
+        string='Mail ID (tech)',
+        help='ID of the related mail_mail. This field is an integer field because '
+             'the related mail_mail can be deleted separately from its statistics. '
+             'However the ID is needed for several action and controllers.',
+        index=True)
+    email = fields.Char(string="Email")
+    message_id = fields.Char(string='Message-ID', help="Technical field for the email Message-ID (RFC 2392)")
+    # document
+    model = fields.Char(string='Document model')
+    res_id = fields.Integer(string='Document ID')
```

Comment:
```suggestion
    res_id = fields.Many2oneReference(string='Document ID', model_field='model')
```
It should be now a good opportunity to apply this.

Reply:
Yes, Good point! I made the change (and rebase). Thank You